lib/Backend/arm64/LowerMD.cpp

```diff
@@ -2089,9 +2090,30 @@ LowererMD::ChangeToAssign(IR::Instr * instr, IRType type)
             instr->ReplaceSrc1(instr->GetSrc1()->UseWithNewType(TyFloat64, instr->m_func));
         }
     }
+    else if ((IRType_IsSignedInt(destType) || IRType_IsUnsignedInt(destType)) && TySize[destType] > TySize[srcType])
+    {
+        // If we're moving between different lengths of registers, we need to use the
+        // right operator - sign extend if the source is int, zero extend if uint.
+        if (IRType_IsSignedInt(srcType))
+        {
+            instr->ReplaceSrc1(src->UseWithNewType(IRType_EnsureSigned(destType), instr->m_func));
+            instr->SetSrc2(IR::IntConstOpnd::New(BITFIELD(0, (TySize[srcType]*8)-1), TyMachReg, instr->m_func, true));
```

Comment:
nit: good to use MachBits

Reply:
MachBits doesn't typecheck, and other uses of BITFIELD are also using TyMachReg.

Reply:
Nvm you're right.